### PR TITLE
search: show indexserver host on index status page

### PR DESCRIPTION
**Pull request by stefanhengl**
It's a common problem to figure out on which indexserver a repo's index resides. With this change we can finally look it up on the index status page.

Screenshot 2022-12-14 at 12 24 53

Note that this comes with the same limitation as the "Reindex now" button: The host can only be resolved if the socket connection between webserver and indexserver is working. Right now we know this isn't the case for docker-compose on Mac with VirtioFS enabled. 
Because webserver might retry for a long time to connect to indexserver, we give it a limited budget (500ms) before falling back to showing "unknown". In my tests, where I simulated a broken connection, the 500ms delay still felt responsive.

## Test plan
- manual testing on a local instance with a working and faulty socket connection.

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.00% (0.00 kb) | 0.00% (+0.35 kb) | 0.00% (+0.35 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/sh/display-indexserver/compare-report.html;tab=live_object) for a full comparison between the commits a9c727a and cb377c6 or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

File: cmd/frontend/graphqlbackend/repository_text_search_index.go [hunk 1]
```diff
@@ -67,6 +68,28 @@ func (r *repositoryTextSearchIndexResolver) Status(ctx context.Context) (*reposi
 	return &repositoryTextSearchIndexStatus{entry: *entry}, nil
 }
 
+func (r *repositoryTextSearchIndexResolver) Host(ctx context.Context) (*repositoryIndexserverHostResolver, error) {
+	// We don't want to let the user wait for too long. If the socket
+	// connection is working, 500ms should be generous.
+	ctx, cancel := context.WithDeadline(ctx, time.Now().Add(time.Millisecond*500))
```
Comment (keegancsmith) [inline cmd/frontend/graphqlbackend/repository_text_search_index.go:74]:
```suggestion
	ctx, cancel := context.WithTimeout(ctx, time.Millisecond*500)
```

File: cmd/frontend/graphqlbackend/repository_text_search_index.go [hunk 2]
```diff
@@ -67,6 +68,28 @@ func (r *repositoryTextSearchIndexResolver) Status(ctx context.Context) (*reposi
...
+	defer cancel()
+	host, err := searchzoekt.GetIndexserverHost(ctx, r.repo.RepoName())
+	if err != nil {
+		host = searchzoekt.Host{Name: "unknown"}
```
Comment (keegancsmith) [inline cmd/frontend/graphqlbackend/repository_text_search_index.go:78]: will this not be confusing for places we don't support this? Should we instead just return a nil value and then not display it / give a good message?

File: cmd/frontend/graphqlbackend/schema.graphql [hunk 3]
```diff
@@ -3499,6 +3503,16 @@ type RepositoryTextSearchIndexStatus {
     otherBranchesNewLinesCount: Int!
 }
 
+"""
+Information about the indexserver that hosts the repo's index.
```
Comment (keegancsmith) [inline cmd/frontend/graphqlbackend/schema.graphql:3507]: do you imagine us adding new entries to this? Can you give an example potential new field for this? ie why not just a string

**Review by keegancsmith — APPROVED** [review]

File: cmd/frontend/graphqlbackend/schema.graphql (hunk 3 above)
Comment (stefanhengl) [reply cmd/frontend/graphqlbackend/schema.graphql:3507]: That is mostly just future-proofing. Maybe, at some point, we might want to return 
- ENV vars
- whether it is a performance or efficency indexserver
- health check data
- free disk space
- ...

**Pull request merged**